### PR TITLE
Add support for DigitalOcean Spaces storage

### LINKED ISSUE #835: Implement storage.Backend for DO Spaces
**Is your feature request related to a problem? Please describe.**
Digital Ocean Spaces https://www.digitalocean.com/products/spaces/ is a s3 compatible blob storage with nice pricing. It would be great to be able to use that as modules storage.

**Describe the solution you'd like**
Spaces is s3 compatible and can be used with the minio client we're already using. We could probably use the minio implementation. The only thing to add would be configuration.

**Additional context**
https://developers.digitalocean.com/documentation/spaces/

### PR DESCRIPTION
**Pull request by t-tomalak**
**What is the problem I am trying to address?**

Add support for DigitalOcean Spaces

**How is the fix applied?**

Added Region variable to Minio configuration which is needed for DigitalOcean Spaces but not necessary for Minio storage.

**Mention the issue number it fixes or add the details of the changes if it doesn't have a specific issue.**

Fixes #835

### CODE REVIEW

**Comment by t-tomalak** [comment]
Hi, I don't think there is even good DO Spaces SDK in Go. Also DigitalOcean at their site, as example how to use their storage is using `minio-go` client https://developers.digitalocean.com/documentation/spaces/

**Comment by marwan-at-work** [comment]
huh
>he API is interoperable with Amazon's AWS S3 API allowing you to interact with the service while using the tools you already know.

Does that mean we can just use our s3 package?

**Comment by t-tomalak** [comment]
Minio also interpolate with S3 API, that's why we are using it for tests in s3. But in s3 config we don't provide endpoint and some other stuff which are not needed. That's why is easier to adopt minio configuration

**Comment by t-tomalak** [comment]
I don't think is good place for that discussion but it may be a good idea if we review and decide on one implementation to use to "rule them all". I mean maybe we could remove `storage/minio` or leave there only `minio.go` and wrap in it `s3.New` method which already allow to customize it's configuration `func New(s3Conf *config.S3Config, cdnConf *config.CDNConfig, options ...func(*aws.Config)) (*Storage, error)`.

**Comment by marwan-at-work** [comment]
@t-tomalak just double checking: S3/Spaces/Minio are *all* compatible with each other?

**Comment by t-tomalak** [comment]
@marwan-at-work Yes, but for us they differentiate on what you need to provide in configurations

**Comment by marpio** [comment]
Since the ability to stream is pretty important to us, we should evaluate how it's handled by each package.

**Comment by arschles** [comment]
This would be nice to have for `v0.1.0`

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -222,6 +223,7 @@ func TestParseExampleConfig(t *testing.T) {
 			Secret:    "minio123",
 			EnableSSL: false,
 			Bucket:    "gomods",
+			Region:    "",
```
Comment (manugupt1) [inline pkg/config/config_test.go]: We can remove this as `""` is zero-value.

Comment (manugupt1) [reply pkg/config/config_test.go]: thanks! let us wait for one more review and merge it.

**Pull request merged**